### PR TITLE
HTTP and RestEasy-related content review

### PR DESCRIPTION
**Pull request by MichalMaler**
This is the docs team follow-up of the @sberyozkin 's PR https://github.com/quarkusio/quarkus/pull/36541/files

This PR will need to be rebased when the https://github.com/quarkusio/quarkus/pull/36667 is merged so that we have separated commits for these fixes and fixes only related to the CORS content.

### CODE REVIEW

File: docs/src/main/asciidoc/security-cors.adoc
Comment (jmartisk) [inline docs/src/main/asciidoc/security-cors.adoc:14]: I think the original wording here was just fine, why change it?
- "Mechanism" is surely a better word than system
- the "should" word is important, it's just a recommendation for the web browser - the browser doesn't technically HAVE TO conform to it
- the word "comprising" is also a bit weird here - one origin consists of a single domain, single scheme and a single port

File: docs/src/main/asciidoc/security-cors.adoc
Comment (jmartisk) [inline docs/src/main/asciidoc/security-cors.adoc:27]: Either "enabled" or "active". "Activated" sounds like it can be activated/deactivated dynamically, but this is something that happens just once for the whole application lifecycle.

File: docs/src/main/asciidoc/security-cors.adoc
Comment (jmartisk) [inline docs/src/main/asciidoc/security-cors.adoc:28]:
```suggestion
It will also add headers configured with the following properties before forwarding the request to its intended destination, like a servlet, Jakarta REST resource, or other endpoints.
```

File: docs/src/main/asciidoc/security-cors.adoc
Comment (jmartisk) [inline docs/src/main/asciidoc/security-cors.adoc:44]: I liked the passive voice better here... Might be just me though.

File: docs/src/main/asciidoc/security-cors.adoc
Comment (MichalMaler) [reply docs/src/main/asciidoc/security-cors.adoc:28]: Great one. Applying

**Comment by MichalMaler** [comment]
@jmartisk Thanks for always great and quick review! Applied all your suggestions, squashed the commits and did a rebase. This version should be good now for @sberyozkin final check.

File: docs/src/main/asciidoc/resteasy.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/resteasy.adoc:706]:
```suggestion
Quarkus includes a CORS filter. Read the xref:security-cors.adoc#cors-filter[CORS filter] section of the "Cross-origin resource sharing" guide to learn how to use it.
```

This is what Lint suggests below

File: docs/src/main/asciidoc/security-cors.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-cors.adoc:17]: Good idea :-), but do we know for sure that RFC does not talk about some origins which makes sense in context of CORS ? I don't mind keeping this reference, but I'd only keep in the Rerefences for the curious users to follow if they'd like

**Comment by sberyozkin** [comment]
Thanks @MichalMaler for the PR, LGTM overall, but propose to keep non-CORS related updates in another PR, or at least in a dedicated commit of its own; added a couple of minor enough suggestions related to the CORS content, cheers

File: docs/src/main/asciidoc/security-cors.adoc
Comment (sberyozkin) [reply docs/src/main/asciidoc/security-cors.adoc:14]: Sorry, missed `These origins consist of a single domain, scheme, and port.` is now a more focused, dedicated sentence.

File: docs/src/main/asciidoc/security-cors.adoc
Comment (sberyozkin) [reply docs/src/main/asciidoc/security-cors.adoc:17]: The RFC though is quite technical, but lets keep, I'd only suggest to keep the link in References but I'll let you decide :+1:

**Comment by MichalMaler** [comment]
> Thanks @MichalMaler for the PR, LGTM overall, but propose to keep non-CORS related updates in another PR, or at least in a dedicated commit of its own; added a couple of minor enough suggestions related to the CORS content, cheers

Hello! Sure. Creating a separate PR was quicker for me, so here it is :) https://github.com/quarkusio/quarkus/pull/36667

This PR can be merged after https://github.com/quarkusio/quarkus/pull/36667 is merged and this PR is rebased.

**Comment by sberyozkin** [comment]
Sounds good @MichalMaler, thanks for the patience, cheers

**Comment by sberyozkin** [comment]
Hi @MichalMaler I updated the title a bit now that it is pure HTTP and RestEasy related updates, thanks

**Comment by sberyozkin** [comment]
Please sort out the conflicts

**Review by geoand — APPROVED** [review]

**Review by sberyozkin — APPROVED** [review]
Adding an approval for completeness now that Georgios approved

**Comment by sberyozkin** [comment]
Michal, not sure that the cors doc commit should be in this PR ?

**Comment by MichalMaler** [comment]
> Michal, not sure that the cors doc commit should be in this PR ?

Oh, no worries, that was just a left over of my actions in the past. I squashed the commits and rename it so that it match our current intention and the heading you proposed :)

**Pull request merged**
